tests/test_omegaconf.py

```diff
@@ -328,6 +328,25 @@ def test_is_none(fac: Any, is_none: bool) -> None:
     assert OmegaConf.is_none(cfg, "node") == is_none
 
 
+@pytest.mark.parametrize(
+    ("cfg", "key", "is_none"),
+    [
+        ({"foo": "${none}", "none": None}, "foo", True),
+        ({"foo": "${missing_node}"}, "foo", False),
+        ({"foo": "${missing_resolver:}"}, "foo", False),
+        ({"foo": "${oc.decode:'[1, '}"}, "foo", False),
```

Comment:
I am okay with a breaking change for OmegaConf.is_none() to make those all raise an exception.
1. It's very unlikely that anyone is depending on this.
2. This behavior is not intentional.

If we do that, I think we can get away without deprecating the public OmegaConf.is_none().

Reply:
Just to confirm, this will raise an exception by default with the deprecation plan I have in mind, but users will be able to silence the exception by specifying `throw_on_resolution_failure=False`

Reply:
Resolving since we decided to deprecate it in the end